plugins/gaming.py

```diff
@@ -116,21 +116,89 @@ def dice(text, notice):
         return "{} ({})".format(total, ", ".join(rolls))
 
 
+choice_split_re = re.compile(r'\s*(?:, ?or|,|or)\s*', re.I)  # regex for splitting choices for choose command
+
+
 @hook.command
 def choose(text, event):
     """<choice1>, [choice2], [choice3], etc. - randomly picks one of the given choices
     :type text: str
     """
-    choices = re.findall(r'([^,]+)', text.strip())
+    choices = choice_split_re.split(text.strip())
+
     if len(choices) == 1:
-        choices = choices[0].split(' or ')
-        if len(choices) == 1:
-            event.notice_doc()
-            return
+        event.notice_doc()
+        return
 
     return random.choice([choice.strip() for choice in choices])
 
 
+# array of digits/characters indexed by int value
+val_char_map = [chr(x) for x in
+                [c for r in (range(ord('0'), ord('9') + 1), range(ord('a'), ord('z') + 1)) for c in r]]
+
+
+def order_bounds(value_a, value_b):
+    """Return the input bounds correctly ordered for randint function"""
+    if value_a < value_b:
+        return value_a, value_b
+
+    return value_b, value_a
+
+
+def choose_integer(bound_a, bound_b):
+    """"Picks a random number from the given range"""
+    # parse as integer range
+    bound_a = int(bound_a)
+    bound_b = int(bound_b)
+
+    lower_bound, upper_bound = order_bounds(bound_a, bound_b)
+    return random.randint(lower_bound, upper_bound)
+
+
+def choose_alphanumeric(bound_a, bound_b):
+    """Tries to pick a value from the given 0-Z range"""
+    # get the integer index of the digits/characters
+    bound_a = int(bound_a, 36)
+    bound_b = int(bound_b, 36)
+
+    lower_bound, upper_bound = order_bounds(bound_a, bound_b)
+
+    if lower_bound < 0 or upper_bound > 36:
+        raise ValueError
+
+    choice = random.randint(lower_bound, upper_bound)
+    return str(val_char_map[choice]).upper()
+
+
+@hook.command("chooser", "chooserange")
+def chooserange(text, event, notice):
+    """<a> to <b>|<a> <b> - picks a random integer or character from the given range"""
+    choices = re.split(r'\s*(?:to| +)\s*', text)
```

Comment:
This should require whitespace around the `to`
```suggestion
    choices = re.split(r'\s*(?:\sto\s| +)\s*', text)
```